packages/relay-compiler/bin/RelayCompilerBin.js

```diff
@@ -122,6 +128,9 @@ function getRelayFileWriter(baseDir: string) {
 function getSchema(schemaPath: string): GraphQLSchema {
   try {
     let source = fs.readFileSync(schemaPath, 'utf8');
+    if (path.extname(schemaPath) === '.json') {
+      source = printSchema(buildClientSchema(JSON.parse(source).data));
+    }
```

Comment:
It seems inefficient to build a client schema, print it, and then parse it back to schema again just a few lines later. Lets refactor to avoid this extra conversion, as it could slow down the compiler for apps with larger schemes.